### DISCUSSION TITLE
Georges Hatem discussion

### DISCUSSION THREAD

**Discussion by gcarreno** [ObjectPascal-Community/1brc-ObjectPascal#81] [General]
Hey Georges(@georges-hatem),

This kicks off the discussion about your entry.

Which, unfortunately is now stuck due to an issue with `mORMot2` not compiling on Delphi once you set a Linux 64 target.

Already dropped an issue on the `mORMot2` repository: https://github.com/synopse/mORMot2/issues/246

Let's see what happens.

Cheers,
Gus

**Comment by synopse** [comment]
mORMot does not support Delphi for Linux.
We use FPC to compile for Linux - but the code needs to be FPC and Delphi compatible.

**Comment by gcarreno** [comment]
Hey Georges(@georges-hatem),

Last `PR` hash is good:
```console
$ ./test_all.sh ghatem
******** Test ********

===== Georges Hatem - FPC ======

real	0m5.987s
user	2m26.719s
sys	0m2.403s
4256d19d3e134d79cc6f160d428a1d859ce961167bd01ca528daca8705163910  /home/gcarreno/Programming/1brc-ObjectPascal/results/ghatem.output
4256d19d3e134d79cc6f160d428a1d859ce961167bd01ca528daca8705163910  Official Output Hash
===========
```

Timing with Desktop on:
```console
$ ./run_all.sh ghatem
******** Run ********

===== Georges Hatem - FPC ======
-- SSD --
Benchmark 1: ghatem
  Time (mean ± σ):      6.016 s ±  0.082 s    [User: 145.823 s, System: 2.436 s]
  Range (min … max):    5.880 s …  6.168 s    10 runs
 
===========
```

`hyperfine` file:
[ghatem-1_000_000_000-SSD.json](https://github.com/gcarreno/1brc-ObjectPascal/files/15038689/ghatem-1_000_000_000-SSD.json)

**Comment by gcarreno** [comment]
Hey Georges(@georges-hatem),

Latest `PR` results on a not so quiet system:
```console
$ ./run_all.sh ghatem
******** Run ********

===== Georges Hatem - FPC ======
-- SSD --
Benchmark 1: ghatem
  Time (mean ± σ):      4.191 s ±  0.054 s    [User: 99.295 s, System: 2.275 s]
  Range (min … max):    4.120 s …  4.263 s    10 runs
 
===========
```
And the latest `hyperfine` file:
[ghatem-1_000_000_000-SSD.json](https://github.com/gcarreno/1brc-ObjectPascal/files/15053198/ghatem-1_000_000_000-SSD.json)

**Reply by georges-hatem** [reply]
thanks Gus for the separate results. that's more improvements than expected

**Comment by paweld** [comment]
Hi @georges-hatem 
Your latest version compiled with the `Release` configuration (`Lazarus trunk / FPC trunk`) causes an error: `ERROR: Stream write error`

**Reply by georges-hatem** [reply]
Hello @paweld ,
Right because the 3 configurations to compile are `Hashmod`, `Hashmult` and `Lemire`.

If you prefer that I set it up differently please let me know.

**Reply by paweld** [reply]
It doesn't bother me, but I know that @gcarreno  compiles only the `release` configuration so your version won't pass the tests.

**Reply by georges-hatem** [reply]
Got it. I'll unify it all back under Release once the winner of the 3 becomes clear
